File: testutils/src/main/java/com/mparticle/mock/MockCoreCallbacks.java
package com.mparticle.mock;

import android.app.Activity;
import android.net.Uri;

import com.mparticle.MParticleOptions;
import com.mparticle.internal.CoreCallbacks;

import org.json.JSONArray;

import java.lang.ref.WeakReference;
import java.util.Map;

public class MockCoreCallbacks implements CoreCallbacks {

    @Override
    public boolean isBackgrounded() {
        return false;
    }

    @Override
    public int getUserBucket() {
        return 0;
    }

    @Override
    public boolean isEnabled() {
        return false;
    }

    @Override
    public void setIntegrationAttributes(int kitId, Map<String, String> integrationAttributes) {

    }

    @Override
    public Map<String, String> getIntegrationAttributes(int kitId) {
        return null;
    }

    @Override
    public WeakReference<Activity> getCurrentActivity() {
        return null;
    }

    @Override
    public JSONArray getLatestKitConfiguration() {
        return null;
    }

    @Override
    public MParticleOptions.DataplanOptions getDataplanOptions() {
        return null;
    }

    @Override
    public boolean isPushEnabled() {
        return false;
    }

    @Override
    public String getPushSenderId() {
        return null;
    }

    @Override
    public String getPushInstanceId() {
        return null;
    }

    @Override
    public Uri getLaunchUri() {
        return null;
    }

    @Override
    public String getLaunchAction() {
        return null;
    }

    @Override
    public KitListener getKitListener() {
        return KitListener.EMPTY;
    }
}

File: kits/braze/braze-43/src/test/kotlin/com/mparticle/kits/mocks/MockCoreCallbacks.kt
package com.mparticle.kits.mocks

import android.app.Activity
import android.net.Uri
import com.mparticle.MParticleOptions.DataplanOptions
import com.mparticle.internal.CoreCallbacks
import com.mparticle.internal.CoreCallbacks.KitListener
import org.json.JSONArray
import java.lang.ref.WeakReference

class MockCoreCallbacks : CoreCallbacks {
    override fun isBackgrounded(): Boolean = false

    override fun getUserBucket(): Int = 0

    override fun isEnabled(): Boolean = false

    override fun setIntegrationAttributes(
        kitId: Int,
        integrationAttributes: Map<String, String>,
    ) {}

    override fun getIntegrationAttributes(kitId: Int): Map<String, String>? = null

    override fun getCurrentActivity(): WeakReference<Activity>? = null

    override fun getLatestKitConfiguration(): JSONArray? = null

    override fun getDataplanOptions(): DataplanOptions? = null

    override fun isPushEnabled(): Boolean = false

    override fun getPushSenderId(): String? = null

    override fun getPushInstanceId(): String? = null

    override fun getLaunchUri(): Uri? = null

    override fun getLaunchAction(): String? = null

    override fun getKitListener(): KitListener =
        object : KitListener {
            override fun kitFound(kitId: Int) {}

            override fun kitConfigReceived(
                kitId: Int,
                configuration: String?,
            ) {}

            override fun kitExcluded(
                kitId: Int,
                reason: String?,
            ) {}

            override fun kitStarted(kitId: Int) {}

            override fun onKitApiCalled(
                kitId: Int,
                used: Boolean?,
                vararg objects: Any?,
            ) {}

            override fun onKitApiCalled(
                methodName: String?,
                kitId: Int,
                used: Boolean?,
                vararg objects: Any?,
            ) {}
        }
}
